File: DAY 6/findHCFEuclideanAlgo.c
#include<stdio.h>
#include<stdlib.h>
#define ULL unsigned long long int

ULL findHCF(ULL a,ULL b){
    return b==0 ? a:findHCF(b,a%b); // Euclidean algorithm
}

int main()
{
    int N;
    scanf("%d",&N);
    ULL hcf,currNum;
    scanf("%llu",&hcf); //first input as HCF
    for(int ctr=2;ctr<=N;ctr++){
        scanf("%llu",&currNum);
        hcf=findHCF(hcf,currNum);
    }
    printf("%llu",hcf);
}

File: DAY 6/findHCFEuclideanAlgo.java
import java.util.Scanner;

public class findHCFEuclideanAlgo {
    private static int findHCF(int a, int b) {
        return b==0?a:findHCF(b, a%b);
    }
    public static void main(String[] args) {
        int N = 4;
        Scanner in = new Scanner(System.in);
        int hcf = in.nextInt();
        for (int ctr = 2; ctr <= N; ctr++) {
            int currNum = in.nextInt(); 
            hcf = findHCF(hcf, currNum);
        }
        System.out.println(hcf
        );
    }
}